### PR TITLE
Add support for Concat operations

### PR DESCRIPTION
**Pull request by ramakota-helprack**
Pending:
  Test cases.
   I will add test cases for these 2 operations soon.

### CODE REVIEW

File: src/Conversion/ONNXToTorch/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,9 @@ add_onnx_mlir_library(OMONNXToTorch
   NN/GlobalAveragePool.cpp
   NN/ReduceMean.cpp
   NN/Gemm.cpp
+  NN/SoftmaxOp.cpp
+  NN/ConcatOp.cpp
+  NN/AbsOp.cpp
```
Comment (vasantha-helprack) [inline src/Conversion/ONNXToTorch/CMakeLists.txt:19]: @ramakota-helprack, please remove `abs`. I've added it as part of the template PR.

Comment (ramakota-helprack) [reply src/Conversion/ONNXToTorch/CMakeLists.txt]: Resolved

File: src/Conversion/ONNXToTorch/NN/ConcatOp.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	inputs[i].getType().cast<TensorType>();
+      auto inputTy = Torch::ValueTensorType::get(context, 
+	inputTensorType.getShape(), inputTensorType.getElementType());
+      auto inputTorchTensor  = 
+	rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(
+		      loc, inputTy, inputs[i]);
+      inputArrayValues.push_back(inputTorchTensor);
+    }
+    Value inputShapeList = rewriter.create<PrimListConstructOp>(loc, 
+	Torch::ListType::get(inputArrayValues.front().getType()),
```
Comment (oowekyala) [inline src/Conversion/ONNXToTorch/NN/ConcatOp.cpp:88]: I think this type is supposed to be the result type. I would have expected it to either be computed from the entire `inputArrayValues`, or to be `resultTy` (also eliminating the cast op below).

Could you provide some tests that show that this works? If it does, please document your reasoning for generating the code like this in a comment

File: src/Conversion/ONNXToTorch/NN/SoftmaxOp.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    TensorType inputTensorType  = input.getType().cast<TensorType>();
+    auto inputTy = Torch::ValueTensorType::get(context,
+	inputTensorType.getShape(), inputTensorType.getElementType());
+
+    auto inputTorchTensor  = 
+	rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>( 
+			loc, inputTy, input);
+    Value constAxisValue = rewriter.create<ConstantIntOp>(loc,axis);
+    auto resultTy = Torch::ValueTensorType::get(op1.getContext(), 
+	opTensorType.getShape(), opTensorType.getElementType());
```
Comment (oowekyala) [inline src/Conversion/ONNXToTorch/NN/SoftmaxOp.cpp:82]: This is also a dup that could be solved with cfe1328d18d6b6908398e6307ec06dbc3e6f69f2

File: src/Conversion/ONNXToTorch/NN/ConcatOp.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    std::vector<Value> inputArrayValues;
+    for (unsigned int i = 0; i < inputs.size(); i++)
+    {
+      TensorType inputTensorType =
+	inputs[i].getType().cast<TensorType>();
+      auto inputTy = Torch::ValueTensorType::get(context, 
+	inputTensorType.getShape(), inputTensorType.getElementType());
+      auto inputTorchTensor  = 
+	rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(
+		      loc, inputTy, inputs[i]);
```
[the remark this answers is not in the document]
Comment (ramakota-helprack) [reply src/Conversion/ONNXToTorch/NN/ConcatOp.cpp:84]: Resolved in #16

File: src/Conversion/ONNXToTorch/NN/ConcatOp.cpp (hunk 2 above)
Comment (ramakota-helprack) [reply src/Conversion/ONNXToTorch/NN/ConcatOp.cpp:88]: Added tests and added comments in #16

File: src/Conversion/ONNXToTorch/NN/SoftmaxOp.cpp (hunk 3 above)
Comment (ramakota-helprack) [reply src/Conversion/ONNXToTorch/NN/SoftmaxOp.cpp]: Resolved in #16

**Comment by ramakota-helprack** [comment]
Created new PR for Concat operation #16

**Pull request closed without merging**